Mathlib/Order/Interval/Set/Disjoint.lean

```diff
@@ -65,6 +65,18 @@ theorem Ioc_disjoint_Ioi (h : b ≤ c) : Disjoint (Ioc a b) (Ioi c) :=
 theorem Ioc_disjoint_Ioi_same : Disjoint (Ioc a b) (Ioi b) :=
   Ioc_disjoint_Ioi le_rfl
 
+@[simp] theorem Ioi_disjoint_Iio (h : a ≤ b) : Disjoint (Ioi b) (Iio a) :=
+  disjoint_left.mpr fun _ hx hy ↦ not_lt_of_le h (hx.trans hy)
+
+theorem Ioi_disjoint_Iio_self : Disjoint (Ioi a) (Iio a) :=
```

Comment:
This *should* be marked `@[simp]`, but also the name should be `same`, not `self`, to match the theorems above.

(likewise for the other `self` lemma below)